resources/js/components/RegisterComponent.vue

```diff
@@ -0,0 +1,293 @@
+                                    />
+                                    <span
+                                        class="invalid-feedback"
+                                        :class="{ block: password.isValid }"
+                                        role="alert"
+                                    >
+                                        <strong>{{ password.msg }}</strong>
+                                    </span>
+                                </div>
+                            </div>
+
+                            <div class="form-group row">
+                                <label
+                                    for="password-confirm"
+                                    class="col-md-4 col-form-label text-md-right"
+                                    >Confirm Password</label
+                                >
+
+                                <div class="col-md-6">
+                                    <input
+                                        id="password-confirm"
+                                        type="password"
+                                        class="form-control"
+                                        name="password_confirmation"
+                                        required
+                                        autocomplete="new-password"
+                                        v-model="password_confirmation.val"
+                                    />
+                                    <span
+                                        class="invalid-feedback"
+                                        :class="{
+                                            block: password_confirmation.isValid
+                                        }"
+                                        role="alert"
+                                    >
+                                        <strong>{{
+                                            password_confirmation.msg
+                                        }}</strong>
+                                    </span>
+                                </div>
+                            </div>
+
+                            <div class="form-group row mb-0">
+                                <label
+                                    for="password-confirm"
+                                    class="col-md-4 col-form-label text-md-right"
+                                ></label>
+                                <div class="col-md-6 offset-md-4">
+                                    <button class="btn btn-primary pull-left">
+                                        Register
+                                    </button>
+                                </div>
+                            </div>
+                        </form>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+<script>
+export default {
+    data() {
+        return {
+            name: {
+                val: "",
```

Comment:
Nopes, we don't want the message and validation information in this array. They should be separate array. The `data()` should have one object for user data i.e. `user: {name: '', email: '', ...}` and validation data should be stored in a separate `errors` object.